### ISSUE TITLE
[question] armv8 cross compile build failed usins same cananfile.txt used in linux X86_64

### ISSUE THREAD

**Issue by kismetro** [conan-io/conan#13655]
### What is your question?

We used conanfile.txt for intel based build system for several months based on conan v1.59.0.

One of additional requirements from out customer is cross build environment. So we just added host-build profile for armv8, but build failed.

Here is our conanfile.txt:
```
[requires]
zlib/1.2.13
eigen/3.3.7
boost/1.73.0
openssl/1.1.1t
cpprestsdk/2.10.18
protobuf/3.20.0
grpc/1.38.1
libcurl/7.87.0
libpng/1.6.38
opencv/4.5.5

[generators]
cmake_find_package
cmake_paths
CMakeDeps
```

Also below command worked well.
```
conan install -r conancenter -pr hostbuild  -s build_type=Debug --build=missing -if ./Debug conanfile.txt
```

But, when I set cross compile environment as below:
```
conan install -r conancenter -pr:b hostbuild -pr:h vpsdev_armv8 -s build_type=Debug --build=missing -if ./Debug conanfile.txt
```
Using below profile for cross-compile env :
```
[settings]
os=Linux
arch=armv8
compiler=gcc
compiler.version=9
compiler.cppstd=17
compiler.libcxx=libstdc++11
build_type=Release

[conf]
tools.cmake.cmaketoolchain:generator=Ninja

[buildenv]
CC=/l4t/toolchain/bin/aarch64-linux-gcc
CXX=/l4t/toolchain/bin/aarch64-linux-g++
LD=/l4t/toolchain/bin/aarch64-linux-ld
```

overriding some package of conanfile.txt was not worked.
```
WARN: cpprestsdk/2.10.18: requirement boost/1.80.0 overridden by your conanfile to boost/1.73.0
WARN: cpprestsdk/2.10.18: requirement openssl/1.1.1s overridden by your conanfile to openssl/1.1.1t
WARN: websocketpp/0.8.2: requirement boost/1.81.0 overridden by cpprestsdk/2.10.18 to boost/1.73.0
WARN: grpc/1.38.1: requirement openssl/1.1.1n overridden by your conanfile to openssl/1.1.1t
WARN: grpc/1.38.1: requirement zlib/1.2.12 overridden by your conanfile to zlib/1.2.13
WARN: opencv/4.5.5: requirement libpng/1.6.39 overridden by your conanfile to libpng/1.6.38
WARN: opencv/4.5.5: requirement eigen/3.3.9 overridden by your conanfile to eigen/3.3.7
WARN: libtiff/4.4.0: requirement libwebp/1.3.0 overridden by opencv/4.5.5 to libwebp/1.2.4
WARN: protobuf/3.20.0: requirement zlib/1.2.13 overridden by grpc/1.38.1 to zlib/1.2.12
ERROR: protobuf/3.20.0: Incompatible requirements obtained in different evaluations of 'requirements'
    Previous requirements: [zlib/1.2.13]
    New requirements: [zlib/1.2.12]
```

We also tried to use not only conanfile.py but also conan v2. In case of conanfile.py caused same error - cannot override package-. Many packages didnt support v2. So, I decided that we must use conan v1.

In summary, let me know how to handle "Incompatible requirements obtained in different evaluations of 'requirements' " in cross-compile environment using v1.

Thank you in advanced. 

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @kismetro 

Thanks for your report.
First, the same warnings about using incompatible generators that I commented in https://github.com/conan-io/conan/issues/13658#issuecomment-1501748459, also applies here, do not use ``cmake_find_package`` and ``cmake_paths`` together with ``CMakeDeps``. Also, use ``CMakeToolchain`` too.

Then, even if some recipes are not fully 2.0 ready, this is a work in progress and lots of efforts are being done on it, so it would be much higher priority to make sure this works fine in Conan 2.0.
This would be the necessary syntax for Conan 2.0 to define the right overrides, assuming that your consumer directly depends on ``boost`` and ``openssl``. If it doesn't depend on those directly, then it would be ``override=True`` instead of ``force=True``.

```python
from conan import ConanFile


class app(ConanFile):
    name = "my-app"

    settings = "os", "compiler", "build_type", "arch"
    generators = "CMakeDeps", "CMakeToolchain"
 
    def requirements(self):
        self.requires("zlib/1.2.13")
        self.requires("eigen/3.3.7")
        self.requires("boost/1.73.0", force=True)
        self.requires("openssl/1.1.1t", force=True)
        self.requires("cpprestsdk/2.10.18")
        self.requires("protobuf/3.20.0")
        self.requires("grpc/1.38.1")
        self.requires("libcurl/7.87.0")
        self.requires("libpng/1.6.38")
        self.requires("opencv/4.5.5")
```
With this, I am able to pass the "conflict" issue, and got to ``Recipe 'grpc/1.38.1' seems broken.`` one.

**Comment by kismetro** [comment]
Thank you for support!

Q1.
I tried to use your commented python with conan v2. To avoid broken stream, I use grpc 1.50.1 instead of 1.38.1 with protobuf 3.20.0. Protobuf version is important for us because we must use generated cpp files from that version. 

```
class app(ConanFile):
    settings = "os", "compiler", "build_type", "arch"
    generators = "CMakeDeps", "CMakeToolchain"

    build_policy = "missing"

    def requirements(self):        
        #self.requires("grpc/1.38.1")
        self.requires("zlib/1.2.13")
        self.requires("eigen/3.3.7")
        self.requires("boost/1.73.0", force=True)
        self.requires("openssl/1.1.1t", force=True)
        self.requires("cpprestsdk/2.10.18")        
        self.requires("protobuf/3.20.0") or with force or with override
        self.requires("grpc/1.50.1")
        self.requires("libcurl/7.87.0")
        self.requires("libpng/1.6.38")
        self.requires("opencv/4.5.5")
```

Above python is not working because of protobuf version conflict with other dependencies. Is there more comments to solve this version conflict?
```
ERROR: Version conflict: googleapis/cci.20221108->protobuf/3.21.9, grpc/1.50.1->protobuf/3.21.4.
```

Q2.
In addition, I tried to use conan 1.59.0 by using your python class. As expected, I get the same error as below:
~~~
ERROR: protobuf/3.20.0: Incompatible requirements obtained in different evaluations of 'requirements'
    Previous requirements: [zlib/1.2.13]
    New requirements: [zlib/1.2.12]
~~~

But, with grpc 1.50.1 and conan v1 make other error related to cross compile options.
~~~
    def requirements(self):        
        #self.requires("grpc/1.38.1")
        self.requires("zlib/1.2.13")
        self.requires("eigen/3.3.7")
        self.requires("boost/1.73.0", force=True)
        self.requires("openssl/1.1.1t", force=True)
        self.requires("cpprestsdk/2.10.18")        
        self.requires("protobuf/3.20.0")
        self.requires("grpc/1.50.1")
        self.requires("libcurl/7.87.0")
        self.requires("libpng/1.6.38")
        self.requires("opencv/4.5.5")
~~~

Error log : 
~~~
Configuration (profile_host):
[settings]
arch=armv8
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
[options]
[build_requires]
[env]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.system.package_manager:mode=install
[buildenv]
CC=/l4t/toolchain/bin/aarch64-linux-gcc
CXX=/l4t/toolchain/bin/aarch64-linux-g++
LD=/l4t/toolchain/bin/aarch64-linux-ld

Configuration (profile_build):
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
[options]
[build_requires]
[env]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja

conanfile.py: WARN: Recipe 'conanfile.py' is deprecated. Please, consider changing your requirements.
WARN: cpprestsdk/2.10.18: requirement boost/1.80.0 overridden by your conanfile to boost/1.73.0
WARN: cpprestsdk/2.10.18: requirement openssl/1.1.1s overridden by your conanfile to openssl/1.1.1t
WARN: websocketpp/0.8.2: requirement boost/1.81.0 overridden by cpprestsdk/2.10.18 to boost/1.73.0
WARN: grpc/1.50.1: requirement openssl/1.1.1s overridden by your conanfile to openssl/1.1.1t
WARN: grpc/1.50.1: requirement protobuf/3.21.4 overridden by your conanfile to protobuf/3.20.0
WARN: googleapis/cci.20221108: requirement protobuf/3.21.9 overridden by grpc/1.50.1 to protobuf/3.20.0
WARN: grpc-proto/cci.20220627: requirement protobuf/3.21.4 overridden by grpc/1.50.1 to protobuf/3.20.0
WARN: opencv/4.5.5: requirement libpng/1.6.39 overridden by your conanfile to libpng/1.6.38
WARN: opencv/4.5.5: requirement eigen/3.3.9 overridden by your conanfile to eigen/3.3.7
WARN: libtiff/4.4.0: requirement libwebp/1.3.0 overridden by opencv/4.5.5 to libwebp/1.2.4
abseil/20220623.0: WARN: Package binary is corrupted, removing: 502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b
WARN: googleapis/cci.20221108: requirement protobuf/3.21.9 overridden by grpc/1.50.1 to protobuf/3.21.4
conanfile.py: Installing package
Requirements
    abseil/20220623.0 from 'conancenter' - Cache
    ade/0.1.2a from 'conancenter' - Cache
    boost/1.73.0 from 'conancenter' - Cache
    bzip2/1.0.8 from 'conancenter' - Cache
    c-ares/1.18.1 from 'conancenter' - Cache
    cpprestsdk/2.10.18 from 'conancenter' - Cache
    eigen/3.3.7 from 'conancenter' - Cache
    googleapis/cci.20221108 from 'conancenter' - Cache
    grpc/1.50.1 from 'conancenter' - Cache
    grpc-proto/cci.20220627 from 'conancenter' - Cache
    jbig/20160605 from 'conancenter' - Cache
    libbacktrace/cci.20210118 from 'conancenter' - Cache
    libcurl/7.87.0 from 'conancenter' - Cache
    libdeflate/1.17 from 'conancenter' - Cache
    libjpeg/9e from 'conancenter' - Cache
    libpng/1.6.38 from 'conancenter' - Cache
    libtiff/4.4.0 from 'conancenter' - Cache
    libwebp/1.2.4 from 'conancenter' - Cache
    opencv/4.5.5 from 'conancenter' - Cache
    openssl/1.1.1t from 'conancenter' - Cache
    protobuf/3.20.0 from 'conancenter' - Cache
    quirc/1.1 from 'conancenter' - Cache
    re2/20220601 from 'conancenter' - Cache
    websocketpp/0.8.2 from 'conancenter' - Cache
    xz_utils/5.4.0 from 'conancenter' - Cache
    zlib/1.2.13 from 'conancenter' - Cache
    zstd/1.5.4 from 'conancenter' - Cache
Packages
    abseil/20220623.0:502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b - Build
    ade/0.1.2a:502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b - Build
    boost/1.73.0:677bc5d9613d70a62707a5fd6070faad9a968d5d - Build
    bzip2/1.0.8:5eba9f12434cedf0d03ce1024364913afad78727 - Build
    c-ares/1.18.1:00bbef14458338529a7b488fc4eea0bf04ae3ea1 - Build
    cpprestsdk/2.10.18:441ceb65de7a7f159cac084e3249964a7e448976 - Build
    eigen/3.3.7:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    googleapis/cci.20221108:081cebe3882580462f8380f0e50f5ab6bf6d7cd5 - Build
    grpc/1.50.1:a2084b50823b9665eec5612cb7c0b9fa03238f88 - Build
    grpc-proto/cci.20220627:b16df20bea6d826d9eb29e1e39826138f0597888 - Build
    jbig/20160605:8bf12db00cbcc6585ebc73173bd847920a70b790 - Build
    libbacktrace/cci.20210118:c3baf9fae083edda2e0c5ef3337b6a111016a898 - Build
    libcurl/7.87.0:a71faa4f2565aa8373039f4b2327417cbec9e5c2 - Build
    libdeflate/1.17:c3baf9fae083edda2e0c5ef3337b6a111016a898 - Build
    libjpeg/9e:c3baf9fae083edda2e0c5ef3337b6a111016a898 - Build
    libpng/1.6.38:31c36682d676f117a16d36736559d0a732542a25 - Build
    libtiff/4.4.0:f1dc98b1644eafd137daeb26da9ca67ee35468c0 - Build
    libwebp/1.2.4:5a29b353925f6f3cdfbfd4a201628c988a2206a5 - Build
    opencv/4.5.5:7a74f9c6236a1a262123e784cc0b37066f6b2773 - Build
    openssl/1.1.1t:c3baf9fae083edda2e0c5ef3337b6a111016a898 - Build
    protobuf/3.20.0:1244cf715c653b2ff0fa89a1e31a7a312798fcaf - Build
    quirc/1.1:a1111a9809291a90c230f043f21dda475658e649 - Build
    re2/20220601:502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b - Build
    websocketpp/0.8.2:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xz_utils/5.4.0:c3baf9fae083edda2e0c5ef3337b6a111016a898 - Build
    zlib/1.2.13:c3baf9fae083edda2e0c5ef3337b6a111016a898 - Build
    zstd/1.5.4:971f42d1180f6b8c2c85cb5918a001178611a824 - Build
Build requirements
    abseil/20220623.0 from 'conancenter' - Cache
    autoconf/2.71 from 'conancenter' - Cache
    automake/1.16.5 from 'conancenter' - Cache
    b2/4.9.3 from 'conancenter' - Cache
    c-ares/1.18.1 from 'conancenter' - Cache
    cmake/3.25.3 from 'conancenter' - Cache
    googleapis/cci.20221108 from 'conancenter' - Cache
    grpc/1.50.1 from 'conancenter' - Cache
    grpc-proto/cci.20220627 from 'conancenter' - Cache
    libtool/2.4.7 from 'conancenter' - Cache
    m4/1.4.19 from 'conancenter' - Cache
    openssl/1.1.1s from 'conancenter' - Cache
    pkgconf/1.9.3 from 'conancenter' - Cache
    protobuf/3.21.4 from 'conancenter' - Cache
    protobuf/3.21.9 from 'conancenter' - Cache
    re2/20220601 from 'conancenter' - Cache
    zlib/1.2.13 from 'conancenter' - Cache
Build requirements packages
    abseil/20220623.0:8646734ee3e7fd5a9de3513cba5a40bc700a885d - Cache
    autoconf/2.71:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    automake/1.16.5:258f6f05ca54813b3d7180161753df7f9ccf7e1d - Cache
    b2/4.9.3:4db1be536558d833e52e862fd84d64d75c2b3656 - Cache
    c-ares/1.18.1:c69c80d84bcbcc1a9610878ad7c37d88e4806c6d - Cache
    cmake/3.25.3:4db1be536558d833e52e862fd84d64d75c2b3656 - Cache
    googleapis/cci.20221108:322914826f543c22a326334116fb147106a8a1a8 - Build
    grpc/1.50.1:6f51d5f20374c7636c9bd35ab1ed105b95dbada2 - Build
    grpc-proto/cci.20220627:95db3a4fa5543efb7bfb6d63bb91bba209adc659 - Build
    libtool/2.4.7:445047bf696d55be0de53e5d97c24a63a4218039 - Cache
    m4/1.4.19:24647d9fe8ec489125dfbae4b3ebefaf7581674c - Cache
    openssl/1.1.1s:6af9cc7cb931c5ad942174fd7838eb655717c709 - Cache
    pkgconf/1.9.3:24647d9fe8ec489125dfbae4b3ebefaf7581674c - Cache
    protobuf/3.21.4:41287b3c8d7c0816cad65c947e666d590f29d2fe - Build
    protobuf/3.21.9:41287b3c8d7c0816cad65c947e666d590f29d2fe - Build
    re2/20220601:8646734ee3e7fd5a9de3513cba5a40bc700a885d - Build
    zlib/1.2.13:6af9cc7cb931c5ad942174fd7838eb655717c709 - Cache

Cross-build from 'Linux:x86_64' to 'Linux:armv8'
Installing (downloading, building) binaries...
abseil/20220623.0: WARN: Build folder is dirty, removing it: /root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b
abseil/20220623.0: Copying sources to build folder
abseil/20220623.0: Building your package in /root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b
abseil/20220623.0: Generator txt created conanbuildinfo.txt
abseil/20220623.0: Calling generate()
abseil/20220623.0: Preset 'release' added to CMakePresets.json. Invoke it manually using 'cmake --preset release'
abseil/20220623.0: If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G Ninja -DCMAKE_TOOLCHAIN_FILE=/root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/build/Release/generators/conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0067=NEW -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release'
abseil/20220623.0: Aggregating env generators
abseil/20220623.0: Calling build()
abseil/20220623.0: Apply patch (portability): link libm to absl string
abseil/20220623.0: Apply patch (portability): Workaround bug in GCC 7.2
abseil/20220623.0: CMake command: cmake -G "Ninja" -DCMAKE_TOOLCHAIN_FILE="/root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/root/.conan/data/abseil/20220623.0/_/_/package/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b" -DCMAKE_POLICY_DEFAULT_CMP0067="NEW" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/src"
-- Using Conan toolchain: /root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: C++ Standard 17 with extensions OFF
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The CXX compiler identification is GNU 9.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Performing Test USE_STD_STRING_VIEW
-- Performing Test USE_STD_STRING_VIEW - Success
-- Performing Test USE_STD_ANY
-- Performing Test USE_STD_ANY - Success
-- Performing Test USE_STD_OPTIONAL
-- Performing Test USE_STD_OPTIONAL - Success
-- Performing Test USE_STD_VARIANT
-- Performing Test USE_STD_VARIANT - Success
-- Configuring done
-- Generating done
-- Build files have been written to: /root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/build/Release
abseil/20220623.0: CMake command: cmake --build "/root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/build/Release" '--' '-j20'
[38/185] Building CXX object absl/random/CMakeFiles/random_internal_randen_hwaes.dir/internal/randen_detect.cc.o
FAILED: absl/random/CMakeFiles/random_internal_randen_hwaes.dir/internal/randen_detect.cc.o
/usr/bin/c++   -I/root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/src -O3 -DNDEBUG -fPIC   -Wall -Wextra -Wcast-qual -Wconversion-null -Wformat-security -Wmissing-declarations -Woverlength-strings -Wpointer-arith -Wundef -Wunused-local-typedefs -Wunused-result -Wvarargs -Wvla -Wwrite-strings -DNOMINMAX -march=armv8-a+crypto -std=c++17 -MD -MT absl/random/CMakeFiles/random_internal_randen_hwaes.dir/internal/randen_detect.cc.o -MF absl/random/CMakeFiles/random_internal_randen_hwaes.dir/internal/randen_detect.cc.o.d -o absl/random/CMakeFiles/random_internal_randen_hwaes.dir/internal/randen_detect.cc.o -c /root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/src/absl/random/internal/randen_detect.cc
cc1plus: error: bad value (‘armv8-a+crypto’) for ‘-march=’ switch
cc1plus: note: valid arguments to ‘-march=’ switch are: nocona core2 nehalem corei7 westmere sandybridge corei7-avx ivybridge core-avx-i haswell core-avx2 broadwell skylake skylake-avx512 cannonlake icelake-client icelake-server cascadelake tigerlake bonnell atom silvermont slm goldmont goldmont-plus tremont knl knm x86-64 eden-x2 nano nano-1000 nano-2000 nano-3000 nano-x2 eden-x4 nano-x4 k8 k8-sse3 opteron opteron-sse3 athlon64 athlon64-sse3 athlon-fx amdfam10 barcelona bdver1 bdver2 bdver3 bdver4 znver1 znver2 btver1 btver2 native
[57/185] Building CXX object absl/flags/CMakeFiles/flags_parse.dir/parse.cc.o
ninja: build stopped: subcommand failed.
abseil/20220623.0:
abseil/20220623.0: ERROR: Package '502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b' build failed
abseil/20220623.0: WARN: Build folder /root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/build/Release
ERROR: abseil/20220623.0: Error in build() method, line 122
	cmake.build()
	ConanException: Error 1 while executing cmake --build "/root/.conan/data/abseil/20220623.0/_/_/build/502b44fc9097bd3a7fa5e4ac6ee89a691984bb8b/build/Release" '--' '-j20'
~~~

command is here:
~~~
conan install -r conancenter -pr:b vpsdev_x86_64 -pr:h vpsdev_armv8 -s build_type=Release -c tools.system.package_manager:mode=install -b missing -if ./Release conanfile.py
~~~

what did i miss?
To resolve this conflict, conan version doesn`t matter for me.

**Comment by memsharded** [comment]
Quick note:
- `` build_policy = "missing"`` shouldn't be used in most cases. It is intended for special packages that build drivers or very low level stuff, which will only work in the machine it is being built. For most regular packages, the command line ``--build=missing``  should be used.

**Comment by memsharded** [comment]
Defining this set of requirements seems to be good to avoid conflicts in 2.0:
```python
    def requirements(self):        
        #self.requires("grpc/1.38.1")
        self.requires("zlib/1.2.13")
        self.requires("eigen/3.3.7", force=True)
        self.requires("boost/1.73.0", force=True)
        self.requires("openssl/1.1.1t", force=True)
        self.requires("cpprestsdk/2.10.18")        
        self.requires("protobuf/3.20.0", force=True)
        self.requires("grpc/1.50.1")
        self.requires("libcurl/7.87.0")
        self.requires("libpng/1.6.38", force=True)
        self.requires("opencv/4.5.5")
        self.requires("libwebp/1.3.0", override=True)
```
Mixing older versions of packages like ``boost/1.73`` with more modern versions of other packages makes necessary to solve a few conflicts up the graph.

However this still cannot progress because the ffmpeg recipe hasn't been updated to 2.0 yet.

The error in 1.X seems unsolvable at this stage, it would be extremely challenging and with very high risk of breaking existing users, so it would be necessary to wait a bit until recipes in ConanCenter are fully updated to 2.0.

**Comment by kismetro** [comment]
Thank you.

I use conan v2 with these requirements : 
(boost from 1.73.0 -> 1.81.0)
~~~
def requirements(self):        
        self.requires("zlib/1.2.13")
        self.requires("eigen/3.3.7", force=True)
        self.requires("boost/1.81.0", force=True)
        self.requires("openssl/1.1.1t", force=True)
        self.requires("cpprestsdk/2.10.18")        
        self.requires("protobuf/3.20.0", force=True)
        self.requires("grpc/1.50.1")
        self.requires("libcurl/7.87.0")
        self.requires("libpng/1.6.38", force=True)
        self.requires("opencv/4.5.5")
        self.requires("libwebp/1.3.0", override=True)
~~~

But version conflict is not solved yet.
Here is my command -> 
~~~
conan install -r conancenter -pr:b vpsdev_host -pr:h vpsdev_armv8 -s build_type=Release -c tools.system.package_manager:mode=install -b missing -of ./Release conanfile.py
~~~

And error messages ->
~~~
======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.system.package_manager:mode=install
[buildenv]
CC=/l4t/toolchain/bin/aarch64-linux-gcc
CXX=/l4t/toolchain/bin/aarch64-linux-g++
LD=/l4t/toolchain/bin/aarch64-linux-ld

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
[conf]
tools.cmake.cmaketoolchain:generator=Ninja


======== Computing dependency graph ========
googleapis/cci.20221108: RUN: cmake --version

googleapis/cci.20221108: RUN: cmake --version

Graph root
    conanfile.py: /opt/workspace/01_Dev/71_client_lib/build/linux/conanPkg/conanfile.py
Requirements
    abseil/20220623.0#ef93150e7eb7891cb99d1dfb0aeeb877 - Cache
    boost/1.81.0#05966cf97c97cdd334623324750f67a0 - Cache
    bzip2/1.0.8#411fc05e80d47a89045edc1ee6f23c1d - Cache
    c-ares/1.18.1#d642963a550bca20b0efc55871771466 - Cache
    cpprestsdk/2.10.18#5ed8ff5504d897c92ac8657cde895f5a - Cache
    eigen/3.3.7#3e0da8490c5baaa197f40586827740a0 - Cache
    googleapis/cci.20221108#e4bebdfa02f3b6f93bae1d5001b8d439 - Cache
    grpc/1.50.1#efcf2d7f5fbe05b3aa99c6159e7f374b - Cache
    grpc-proto/cci.20220627#ddfebfb50349c59afbd3be34ca0f590f - Cache
    libbacktrace/cci.20210118#ec1aa63bbc10145c6a299e68e711670c - Cache
    openssl/1.1.1t#7d9fc949345908e1b8609d9d9f528a08 - Cache
    protobuf/3.20.0#b9e4230fca31f7beb4908d0dcfbfe83e - Cache
    re2/20220601#9194be01ae4edae3c2f4cbbec2c57884 - Cache
    websocketpp/0.8.2#3aed387a9bd2bda5e0c70a04c5e7ea26 - Cache
    zlib/1.2.13#13c96f538b52e1600c40b88994de240f - Cache
Build requirements
    abseil/20220623.0#ef93150e7eb7891cb99d1dfb0aeeb877 - Cache
    b2/4.9.3#5445210661950ce568cfd76158d9e117 - Cache
    c-ares/1.18.1#d642963a550bca20b0efc55871771466 - Cache
    cmake/3.25.3#586c962fa58ccc886a7b2667f7c19ab9 - Cache
    googleapis/cci.20221108#e4bebdfa02f3b6f93bae1d5001b8d439 - Cache
    grpc/1.50.1#efcf2d7f5fbe05b3aa99c6159e7f374b - Cache
    openssl/1.1.1s#3eeebc156917046ee80533dbbb8f4671 - Cache
    protobuf/3.21.4#13e6dd4dd345ba08799ae4ccb00111e1 - Cache
    protobuf/3.21.9#515ceb0a1653cf84363d9968b812d6be - Cache
    re2/20220601#9194be01ae4edae3c2f4cbbec2c57884 - Cache
    zlib/1.2.13#13c96f538b52e1600c40b88994de240f - Cache
ERROR: Version conflict: googleapis/cci.20221108->protobuf/3.21.9, grpc/1.50.1->protobuf/3.21.4.
~~~

Please let me know one more step to resolve this conflict!

**Comment by memsharded** [comment]
Hi  @kismetro 

Sorry this was not followed up back then.
It seems it fell through the cracks, sometimes happen when load is very high.

I am marking this ticket as staled, to be closed if there is no further activity, but please let us know

**Comment by AbrilRBS** [comment]
Closing as stale — @kismetro, we never heard back after memsharded's note in April that this would be marked stale without further activity. Thanks for working through the cross-compilation setup with us in this thread; please open a new issue linking back here if you run into this again on Conan 2.
